src/services/smart-fields-values-injector.js

```diff
@@ -33,6 +33,19 @@ function SmartFieldsValuesInjector(record, modelName, fieldsPerModel) {
         if (_.isFunction(value.then)) {
           return value
             .then((result) => {
+              // NOTICE: We need some recursion in order to consider smart fields in smart
+              //         relations. So if the result contains data and the field has a reference
+              //         then we consider we have to inject smart fields within the reference.
+              if (result && result.dataValues && field.reference) {
+                const [referenceModelName] = field.reference.split('.');
+                const smartFieldsValuesInjector = new SmartFieldsValuesInjector(
+                  result,
+                  referenceModelName,
+                  fieldsPerModel,
```

Comment:
Not really sure what is contained in `fieldsPerModel`.
I am sure you fix works as you requested a review, but I wonder if this will always contain the necessary context.

Reply:
`fieldsPerModel` contains all the requested fields. Exemple:

```
// Calling this on a lumber instance:
http://example.org/forest/collection_2?fields[col_1]=CHID_and_status&fields[collection_2]=aml_status,created_at,deleted_at,col_1,&timezone=Europe/Paris
```

```
// Gives:
{
  col_1: ['CHID_and_status'],
  collection_2: ['aml_status', 'created_at', deleted_at', 'col_1']
}
```